### PR TITLE
Feature/home UI

### PR DESCRIPTION
**Pull request by Soha-Ahmed-Hamdy**
Adding recipes to favourite and make indicator to load until receiving data and handling favourite Button clicks(save and delete)

### CODE REVIEW

File: iOS_Project/Module/Home/View/HomeViewController.swift [hunk 1]
```diff
@@ -43,12 +88,72 @@ class HomeViewController: UIViewController , UITableViewDelegate,UITableViewData
...
+            cell.catergoryImage.backgroundColor = UIColor.orange
+        }else{
+            cell.catergoryImage.backgroundColor = UIColor.white
+        }
+    }
+    
+    func fetchingData(){
+        switch selectedCategory{
+            case "Popular":
+                homeViewModel?.fetchHomeData(tag: "middle_eastern")
```
Comment (MinaKamal99) [inline iOS_Project/Module/Home/View/HomeViewController.swift]: please add a function to ignore the redundancy

File: iOS_Project/Module/Home/View/HomeViewController.swift [hunk 2]
```diff
@@ -43,12 +88,72 @@ class HomeViewController: UIViewController , UITableViewDelegate,UITableViewData
         cell?.setUpCell()
         cell?.catergoryName.text = categoryNames[indexPath.row]
         cell?.catergoryImage.image = UIImage(named: categoryNames[indexPath.row])
+        cell?.catergoryImage.layer.cornerRadius = 15
+        cell?.layer.cornerRadius = 15
```
Comment (MinaKamal99) [inline iOS_Project/Module/Home/View/HomeViewController.swift:92]: please add these variables as enum Constants to be easily modified later on

File: iOS_Project/Module/Home/View/HomeViewController.swift (hunk 1 above)
Comment (Soha-Ahmed-Hamdy) [reply iOS_Project/Module/Home/View/HomeViewController.swift]: comments resolved

**Review by MinaKamal99 — APPROVED** [review]
Thanks 🙏

**Pull request merged**